simulation.cpp

```diff
@@ -197,8 +203,34 @@ void assign_inputs(population &p, const std::vector<double> &inputs)
 {
   std::vector<individual> vec_inds;
   for(auto& ind : p.get_inds()){
-      ind.set_input(inputs);
+      ind.assign_input(inputs);
+    }
+}
+
+bool all_individuals_have_same_input(const simulation &s)
+{
+  population p = s.get_pop();  //Would it be prefered to refer to a all_individuals_have_same_input population-level function instead?
+
+  std::vector<double> input_first_individual = p.get_inds()[0].get_input_values();
+
+  for(auto& ind : p.get_inds()){
+      if(input_first_individual != ind.get_input_values()){
+          return false;
+        }
     }
+  return true;
+}
+
+std::vector<double> get_current_input(const simulation &s)
+{
+ assert(all_individuals_have_same_input(s));
+ return s.get_pop().get_inds()[0].get_input_values();
```

Comment:
too many dots 🎵   too many dots 🎵  Wrap in a function like `get_nth_individual(n_individual)` and/or `get_nth_individual_inputs(n_individual)`